### PR TITLE
Check self-hosted runners are online

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

#18905 checks if the docker could be launched inside the runners.

However, the runners could be offline due to some unknown reasons, and we are not aware of this problem (job hangs forever) so far.

This PR adds a check for runner being online or offline.

However, it might happen that a runner becomes offline in the middle of a workflow run. This situation is not easy to deal with, and we still need to prevent such situation. Therefore, a new scheduled (per hour) workflow is created to check runner availability.

### CODE REVIEW

File: .github/workflows/check_runner_status.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+#
+# For example for `run_all_tests_torch_cuda_extensions_gpu` the docker image is
+# `huggingface/transformers-pytorch-deepspeed-latest-gpu`, which can be found at
+# `docker/transformers-pytorch-deepspeed-latest-gpu/Dockerfile`
+
+on:
+  repository_dispatch:
+  schedule:
+    # run per hour
+    - cron: "* */1 * * *"
```
Comment (ydshieh) [inline .github/workflows/check_runner_status.yml:13]: Check runner availability each hour

File: .github/workflows/check_runner_status.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+      - uses: actions/checkout@v2
+      - uses: actions/download-artifact@v2
+      - name: Send message to Slack
+        env:
+          CI_SLACK_BOT_TOKEN: ${{ secrets.CI_SLACK_BOT_TOKEN }}
+          CI_SLACK_CHANNEL_ID: ${{ secrets.CI_SLACK_CHANNEL_ID }}
+          CI_SLACK_CHANNEL_ID_DAILY: ${{ secrets.CI_SLACK_CHANNEL_ID_DAILY }}
+          CI_SLACK_CHANNEL_DUMMY_TESTS: ${{ secrets.CI_SLACK_CHANNEL_DUMMY_TESTS }}
+          CI_SLACK_REPORT_CHANNEL_ID: ${{ secrets.CI_SLACK_CHANNEL_DUMMY_TESTS }}
```
Comment (ydshieh) [inline .github/workflows/check_runner_status.yml:50]: Should I create a new channel?

File: .github/workflows/check_runner_status.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+          fetch-depth: 2
+
+      - name: Check Runner Status
+        run: python utils/check_self_hosted_runner.py --target_runners single-gpu-ci-runner-docker,multi-gpu-ci-runner-docker,single-gpu-scheduled-ci-runner-docker,multi-scheduled-scheduled-ci-runner-docker --token ${{ secrets.ACCESS_REPO_INFO_TOKEN }}
+
+  send_results:
+    name: Send results to webhook
+    runs-on: ubuntu-latest
+    needs: check_runner_status
+    if: ${{ failure() }}
```
Comment (ydshieh) [inline .github/workflows/check_runner_status.yml:35]: Send report only when some runners are not available.

**Review by LysandreJik — APPROVED** [review]
Sounds good to me!

File: .github/workflows/check_runner_status.yml (hunk 2 above)
Comment (LysandreJik) [reply .github/workflows/check_runner_status.yml:50]: I'd use one of the existing channels

**Pull request merged**